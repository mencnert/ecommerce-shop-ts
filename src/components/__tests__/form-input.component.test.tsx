import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import FormInput from "../form-input/form-input.component";

const onChangeMock = jest.fn();

const props = {
  label: "asdf",
  name: "name",
  type: "email",
  handleChange: onChangeMock,
  value: "value",
  required: false,
};

it("calls onChange method", () => {
  const wrapper = shallow(<FormInput {...props} />);
  wrapper.find("input").simulate("change");
  expect(onChangeMock).toBeCalledTimes(1);
});

it("should match snapshot with label", () => {
  const wrapper = shallow(<FormInput {...props} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("should match snapshot without label", () => {
  const wrapper = shallow(<FormInput {...props} label={""} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("should match snapshot without value", () => {
  const wrapper = shallow(<FormInput {...props} value={""} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
