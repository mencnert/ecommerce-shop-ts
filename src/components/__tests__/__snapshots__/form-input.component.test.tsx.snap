// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with label 1`] = `
<div
  className="group"
>
  <input
    className="form-input"
    name="name"
    onChange={[MockFunction]}
    required={false}
    type="email"
    value="value"
  />
  <label
    className="shrink form-input-label"
  >
    asdf
  </label>
</div>
`;

exports[`should match snapshot without label 1`] = `
<div
  className="group"
>
  <input
    className="form-input"
    name="name"
    onChange={[MockFunction]}
    required={false}
    type="email"
    value="value"
  />
</div>
`;

exports[`should match snapshot without value 1`] = `
<div
  className="group"
>
  <input
    className="form-input"
    name="name"
    onChange={[MockFunction]}
    required={false}
    type="email"
    value=""
  />
  <label
    className=" form-input-label"
  >
    asdf
  </label>
</div>
`;
